NETProvider/src/FirebirdSql.Data.FirebirdClient/FirebirdClient/FbParameterCollection.cs

```diff
@@ -217,12 +217,12 @@ internal int IndexOf(string parameterName, int luckyIndex)
 			var normalizedParameterName = FbParameter.NormalizeParameterName(parameterName);
 			if (luckyIndex != -1 && luckyIndex < _parameters.Count)
 			{
-				if (_parameters[luckyIndex].InternalParameterName.Equals(normalizedParameterName, StringComparison.CurrentCultureIgnoreCase))
+				if (_parameters[luckyIndex].InternalParameterName.Equals(normalizedParameterName, StringComparison.OrdinalIgnoreCase))
 				{
 					return luckyIndex;
 				}
 			}
-			return _parameters.FindIndex(x => x.InternalParameterName.Equals(normalizedParameterName, StringComparison.CurrentCultureIgnoreCase));
+			return _parameters.FindIndex(x => x.InternalParameterName.Equals(normalizedParameterName, StringComparison.OrdinalIgnoreCase));
```

Comment:
Ahh, you took lucky shot. Check http://firebird.1100200.n4.nabble.com/DNET-532-tp4640493p4640508.html .